src/main/java/org/jboss/tools/intellij/componentanalysis/CAIntentionAction.java

```diff
@@ -52,7 +57,19 @@ protected CAIntentionAction(PsiElement element, VulnerabilitySource source, Depe
 
     @Override
     public void invoke(@NotNull Project project, Editor editor, PsiFile file) throws IncorrectOperationException {
-        this.updateVersion(project, editor, file, getRecommendedVersion(this.report));
+        String recommendedVersion = getRecommendedVersion(this.report);
+        this.updateVersion(project, editor, file, recommendedVersion);
+        ApiService apiService = ServiceManager.getService(ApiService.class);
+        SaUtils saUtils = new SaUtils();
+        var telemetryMsg = TelemetryService.instance().action("component_analysis_recommendation_accepted");
+        telemetryMsg.property("package", this.report.getRef().name());
+        telemetryMsg.property("version", recommendedVersion);
+        telemetryMsg.property(ApiService.TelemetryKeys.ECOSYSTEM.toString(), saUtils.determinePackageManagerName(file.getName()));
+        telemetryMsg.property(ApiService.TelemetryKeys.PLATFORM.toString(), System.getProperty("os.name"));
+        telemetryMsg.property(ApiService.TelemetryKeys.MANIFEST.toString(), file.getName());
+        telemetryMsg.property(ApiService.TelemetryKeys.RHDA_TOKEN.toString(), ApiSettingsState.getInstance().rhdaToken);
+        apiService.setRequestProperties(file.getName());
```

Comment:
```suggestion
```
This method should not be called. It is only used when calling `exhort-java-api`. Also delete line 62.

Reply:
Done.